### PR TITLE
chore: Adjustments for triggering experiments

### PR DESCRIPTION
**Pull request by robinholzi**
# Bugfixes and Experiment configs

### CODE REVIEW

File: modyn/supervisor/internal/triggers/performance/decision_policy.py [hunk 1]
```diff
@@ -97,13 +97,21 @@ def evaluate_decision(
         mode: TriggerEvaluationMode,
         method: ForecastingMethod,
     ) -> bool:
-        threshold = performance_tracker.forecast_expected_performance(mode) - self.config.allowed_deviation
+        expected = performance_tracker.forecast_expected_performance(mode)
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/performance/decision_policy.py:100]: as discussed, this call is invalid

Comment (MaxiBoether) [reply modyn/supervisor/internal/triggers/performance/decision_policy.py:100]: what i dont understand still is why in the hindsight mode we need any forecasting (let it be the rolling average). i understood it as "if our accuracy on the evaluation interval deviates too much, we trigger". but apparently something more is going on?

File: modyn/supervisor/internal/triggers/performance/decision_policy.py [hunk 2]
```diff
@@ -97,13 +97,21 @@ def evaluate_decision(
...
+            if self.config.absolute:
+                return deviation >= self.config.deviation
+            return deviation >= self.config.deviation * expected
 
-        return (evaluation_scores[self.config.metric] < threshold) or (
-            performance_tracker.forecast_next_performance(mode) < threshold
+        if self.config.absolute:
+            allowed_absolute_deviation = self.config.deviation
+        else:
+            allowed_absolute_deviation = self.config.deviation * expected
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/performance/decision_policy.py:111]: you could also always calculate this and then use that in hindsight mode as well. might simplify this function a bit

File: modyn/supervisor/internal/triggers/performance/decision_policy.py [hunk 3]
```diff
@@ -97,13 +97,21 @@ def evaluate_decision(
...
+
+        return (deviation >= allowed_absolute_deviation) or (
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/performance/decision_policy.py:113]: i guess you could also remove the "if hindsight" codepath, just have this first thing (which is basically hindsight), and then "if lookahead" set another bool to the result of this (or false if not lookahead). reduces the branches here, had some trouble parsing it at first

**Review by MaxiBoether — APPROVED** [review]
Thanks, changes as mostly discussed :)

File: modyn/supervisor/internal/triggers/performance/decision_policy.py (hunk 1 above)
Comment (robinholzi) [reply modyn/supervisor/internal/triggers/performance/decision_policy.py:100]: With 
```python
forecast_expected_performance(
            mode, method="rolling_average"
        )
```
this is equivalent to saying, if current values deviates more than X % of the rolling average, then trigger. We simply just reuse the averaging from the forecasting method. It's not a true forecast.

Comment (MaxiBoether) [reply modyn/supervisor/internal/triggers/performance/decision_policy.py]: as discussde, let's rename expected to "optimal" performance for this

**Pull request merged**